tests/unit/test_loaders.py

```diff
@@ -87,39 +132,63 @@ def test_default_search_paths(self):
         # with .aws/models.
         home_dir_path = os.path.join('.aws', 'models')
         self.assertTrue(
-            any(p.endswith(home_dir_path) for p in loader.search_paths)
+            any(str(p).endswith(home_dir_path) for p in loader.search_paths)
         )
+        self.assertIsInstance(loader.search_paths, SearchPathList)
 
     def test_can_add_to_search_path(self):
         loader = Loader()
         loader.search_paths.append('mypath')
-        self.assertIn('mypath', loader.search_paths)
+        self.assertIn(pathlib.Path('mypath'), loader.search_paths)
+        self.assertIsInstance(loader.search_paths, SearchPathList)
```

Comment:
This seems like an implementation detail, we know if we add a string and it results in a Path object the conversion is working. Whether that's a SearchPathList isn't actually important to us.